velox/type/DecimalUtil.h

```diff
@@ -147,6 +147,36 @@ class DecimalUtil {
     return static_cast<TOutput>(rescaledValue);
   }
 
+  template <typename TInput, typename TOutput>
+  inline static std::optional<TOutput> rescaleFloating(
+      const TInput inputValue,
+      const int toPrecision,
+      const int toScale) {
+    if (!std::isfinite(inputValue)) {
+      return std::nullopt;
+    }
+    auto toValue =
+        inputValue * static_cast<TInput>(DecimalUtil::kPowersOfTen[toScale]);
+    bool isOverflow = !std::isfinite(toValue);
+    toValue = toValue - std::floor(toValue) >= 0.5 ? std::ceil(toValue)
```

Comment:
Can we use exists function to convert double to int128 or int64? https://github.com/facebookincubator/velox/blob/main/velox/type/Conversions.h#L231

Reply:
I don't think this existing function works in this case. This function has no unit test and we know nothing about this function.

If we use `velox::util::Converter<TypeKind::HUGEINT>::cast((double) kLongDecimalMax)`, it will throw exception like 'Loss of precision during arithmetic conversion'. If we use `velox::util::Converter<TypeKind::HUGEINT, void, true>::cast((double)kLongDecimalMax)`, the return value is definitely wrong.

Could you give me an example about using this function to convert double to int128? @jinchengchenghh

Reply:
It is base function, now it may be used nowhere.
If it has error, we could ask the author for help. Can you help look into this issue? @majetideepak

Reply:
@jinchengchenghh I read the source code of 'velox::util::Converter', refer to https://github.com/facebook/folly/blob/main/folly/Conv.h#L1345, this function is suitable here except it will throw an exception if a double number exceeds the limit of an integral number. I think we could catch this exception here and judge if this casting is overflowed. But I am not sure. Could you give me a favor to check if we could use this function in this way?

Reply:
I think it's OK, and you should use the `truncate` from config https://github.com/facebookincubator/velox/blob/main/velox/expression/CastExpr-inl.h#L255

Reply:
No, we can't. Look at https://github.com/facebookincubator/velox/blob/main/velox/type/Conversions.h#L236, if we use the truncate, and the return value's type is int128_t, it will return `std::numeric_limits<int128_t>::max()` directly. That's the reason why I think If we use `velox::util::Converter<TypeKind::HUGEINT, void, true>::cast((double)kLongDecimalMax)`, the return value is definitely wrong.

Besides, this config is used to cast a floating number to an integral number, but we want to cast a floating number to a decimal number. Although we need to cast a floating number to an integral number here, we don't want to get a truncated value here because we need to know whether the overflow occurs here.